In [79]:
#Import the required Libraries.
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

## Data Cleaning 

### Data Reading & Data Types 

In [146]:
#Read the data in pandas
inp0= pd.read_csv("Attribute+DataSet.csv")
inp1= pd.read_csv("Dress+Sales.csv")

inp0.dtypes
inp1.Dress_ID.value_counts()

inp0.shape

(479, 13)

Dress_ID  29-08-2013  31-08-2013  09-02-2013  09-04-2013  09-06-2013  \
0    1006032852        2114        2274        2491        2660        2727   
1    1212192089         151         275         570         750         813   
2    1190380701           6           7           7           7           8   
3     966005983        1005        1128        1326        1455        1507   
4     876339541         996        1175        1304        1396        1432   
..          ...         ...         ...         ...         ...         ...   
474   990559192           0           0           0          60          62   
475   713391965           0           0           0         560         554   
476   532874347           0           0           0         734         728   
477   655464934           0           0           0         254         259   
478   919930954           0           0           0         538         545   

     09-08-2013  09-10-2013 09-12-2013 14-09-2013  ... 24-09-2013 26-09-2013  \
0          2887        2930       3119       3204  ...       3554     3624.0   
1          1066        1164       1558       1756  ...       2710     2942.0   
2             8           9         10         10  ...         11       11.0   
3          1621        1637       1723       1746  ...       1878     1892.0   
4          1559        1570       1638       1655  ...       2032     2156.0   
..          ...         ...        ...        ...  ...        ...        ...   
474          64          65         67         68  ...         73       74.0   
475         544         537        525        519  ...        400      388.0   
476         726         715        694        690  ...        616      597.0   
477         261         263        268        270  ...        257      256.0   
478         558         563        578        585  ...        628      632.0   

    28-09-2013 30-09-2013  10-02-2013  10-04-2013  10-06-2013  10-08-2013  \
0         3706     3746.0      3795.0      3832.0        3897      3923.0   
1         3258     3354.0      3475.0      3654.0        3911      4024.0   
2           11       11.0        11.0        11.0          11        11.0   
3         1914     1924.0      1929.0      1941.0        1952      1955.0   
4         2252     2312.0      2387.0      2459.0        2544      2614.0   
..         ...        ...         ...         ...         ...         ...   
474         75       75.0        76.0        76.0          77        77.0   
475        360      364.0       372.0       377.0         380       382.0   
476        586      569.0       561.0       555.0         551       546.0   
477        255      254.0       253.0       250.0         249       249.0   
478        639      645.0       651.0       655.0         660       668.0   

     10-10-2013  10-12-2013  
0        3985.0        4048  
1        4125.0        4277  
2          11.0          11  
3        1959.0        1963  
4        2693.0        2736  
..          ...         ...  
474        77.0          77  
475       384.0         285  
476       535.0         520  
477       249.0         248  
478       674.0         680  

[479 rows x 24 columns]

You have “Attribute DataSet” which contains a column named “Price”. Choose the correct statement from the following about its data type and variable type.
- Integer type and numerical variable
- Object type and categorical ordinal variable
- Object type and categorical nominal variable
- Float type and categorical variable.


There is another column in “Attribute DataSet” named as “Recommendation”, choose the correct statement about its data type and variable type.
- Integer type and categorical
- Object type and categorical
- Integer type and continuous numerical
- Object type only.


Which of the following column do you think are of no use in “Attribute DataSet”.
- Dress_ID
- Price
- Size and material
- NeckLine
- None of the above


In [12]:
# Print the information about the attributes of inp0 and inp1.

inp0.info()

inp1.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 479 entries, 0 to 478
Data columns (total 13 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   Dress_ID        479 non-null    int64  
 1   Style           479 non-null    object 
 2   Price           477 non-null    object 
 3   Rating          479 non-null    float64
 4   Size            479 non-null    object 
 5   Season          477 non-null    object 
 6   NeckLine        476 non-null    object 
 7   SleeveLength    477 non-null    object 
 8   Material        360 non-null    object 
 9   FabricType      223 non-null    object 
 10  Decoration      255 non-null    object 
 11  Pattern Type    377 non-null    object 
 12  Recommendation  479 non-null    int64  
dtypes: float64(1), int64(2), object(10)
memory usage: 48.8+ KB
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 479 entries, 0 to 478
Data columns (total 24 columns):
 #   Column      Non-Null Count  Dtype  
---  -----

### Fixing the Rows and Columns 

As you can see, there is a column in “Attribute Dataset” named as ‘Size’. This column contains the values in abbreviation format. Write a code in Python to convert the followings:

- M into  “Medium”
- L into  “Large”
- XL into “Extra large”
- free into “Free”
- S, s & small into “Small”.

Now once you are done with changes in the dataset, what is the value of the lowest percentage, the highest percentage and the percentage of Small size categories in the column named “Size”?


In [16]:
inp0.head()

Dress_ID    Style    Price  Rating Size  Season NeckLine SleeveLength  \
0  1006032852     Sexy      Low     4.6    M  Summer   o-neck    sleevless   
1  1212192089   Casual      Low     0.0    L  Summer   o-neck        Petal   
2  1190380701  vintage     High     0.0    L  Automn   o-neck         full   
3   966005983    Brief  Average     4.6    L  Spring   o-neck         full   
4   876339541     cute      Low     4.5    M  Summer   o-neck    butterfly   

        Material FabricType  Decoration Pattern Type  Recommendation  
0            NaN    chiffon     ruffles       animal               1  
1     microfiber        NaN     ruffles       animal               0  
2       polyster        NaN         NaN        print               0  
3           silk    chiffon  embroidary        print               1  
4  chiffonfabric    chiffon         bow          dot               0

In [20]:
# Column fixing, correcting size abbreviation. count the percentage of each size category in "Size" column.

inp0.Size.value_counts(normalize = True)

M        0.356994
free     0.344468
L        0.194154
S        0.070981
XL       0.029228
small    0.002088
s        0.002088
Name: Size, dtype: float64

In [21]:
# Print the value counts of each category in "Size" column.

print(inp0.Size.value_counts(normalize = True))

M        0.356994
free     0.344468
L        0.194154
S        0.070981
XL       0.029228
small    0.002088
s        0.002088
Name: Size, dtype: float64


### Impute/Remove Missing values

In [25]:
# Print the null count of each variables of inp0 and inp1.

inp1.head()
inp1.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 479 entries, 0 to 478
Data columns (total 24 columns):
 #   Column      Non-Null Count  Dtype  
---  ------      --------------  -----  
 0   Dress_ID    479 non-null    float64
 1   29-08-2013  479 non-null    int64  
 2   31-08-2013  479 non-null    int64  
 3   09-02-2013  479 non-null    int64  
 4   09-04-2013  479 non-null    int64  
 5   09-06-2013  479 non-null    int64  
 6   09-08-2013  479 non-null    int64  
 7   09-10-2013  479 non-null    int64  
 8   09-12-2013  479 non-null    object 
 9   14-09-2013  479 non-null    object 
 10  16-09-2013  479 non-null    object 
 11  18-09-2013  479 non-null    object 
 12  20-09-2013  479 non-null    object 
 13  22-09-2013  479 non-null    object 
 14  24-09-2013  479 non-null    int64  
 15  26-09-2013  257 non-null    float64
 16  28-09-2013  479 non-null    int64  
 17  30-09-2013  222 non-null    float64
 18  10-02-2013  220 non-null    float64
 19  10-04-2013  221 non-null    f

You are given another dataset named “Dress Sales”. Now if you observe the datatypes of the columns using ‘inp1.info()’ command, you can identify that there are certain columns defined as object data type though they primarily consist of numeric data.

Now if you try and convert these object data type columns into numeric data type(float), you will come across an error message. Try to correct this error.







In [37]:
inp1.dtypes

inp1.astype(float)

ValueError: could not convert string to float: 'Removed'

In [41]:
# Print the data types information of inp1 i.e. "Dress Sales" data.

inp1.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 479 entries, 0 to 478
Data columns (total 24 columns):
 #   Column      Non-Null Count  Dtype  
---  ------      --------------  -----  
 0   Dress_ID    479 non-null    float64
 1   29-08-2013  479 non-null    int64  
 2   31-08-2013  479 non-null    int64  
 3   09-02-2013  479 non-null    int64  
 4   09-04-2013  479 non-null    int64  
 5   09-06-2013  479 non-null    int64  
 6   09-08-2013  479 non-null    int64  
 7   09-10-2013  479 non-null    int64  
 8   09-12-2013  479 non-null    object 
 9   14-09-2013  479 non-null    object 
 10  16-09-2013  479 non-null    object 
 11  18-09-2013  479 non-null    object 
 12  20-09-2013  479 non-null    object 
 13  22-09-2013  479 non-null    object 
 14  24-09-2013  479 non-null    int64  
 15  26-09-2013  257 non-null    float64
 16  28-09-2013  479 non-null    int64  
 17  30-09-2013  222 non-null    float64
 18  10-02-2013  220 non-null    float64
 19  10-04-2013  221 non-null    f

In [43]:
# Try to convert the object type into float type of data. YOU GET ERROR MESSAGE.

inp1.astype(float)

ValueError: could not convert string to float: 'Removed'

In [47]:
# Do the required changes in the "Dress Sales" data set to get null values on string values.

inp1[inp1.isna()]

Dress_ID  29-08-2013  31-08-2013  09-02-2013  09-04-2013  09-06-2013  \
0         NaN         NaN         NaN         NaN         NaN         NaN   
1         NaN         NaN         NaN         NaN         NaN         NaN   
2         NaN         NaN         NaN         NaN         NaN         NaN   
3         NaN         NaN         NaN         NaN         NaN         NaN   
4         NaN         NaN         NaN         NaN         NaN         NaN   
..        ...         ...         ...         ...         ...         ...   
474       NaN         NaN         NaN         NaN         NaN         NaN   
475       NaN         NaN         NaN         NaN         NaN         NaN   
476       NaN         NaN         NaN         NaN         NaN         NaN   
477       NaN         NaN         NaN         NaN         NaN         NaN   
478       NaN         NaN         NaN         NaN         NaN         NaN   

     09-08-2013  09-10-2013 09-12-2013 14-09-2013  ... 24-09-2013 26-09-2013  \
0           NaN         NaN        NaN        NaN  ...        NaN        NaN   
1           NaN         NaN        NaN        NaN  ...        NaN        NaN   
2           NaN         NaN        NaN        NaN  ...        NaN        NaN   
3           NaN         NaN        NaN        NaN  ...        NaN        NaN   
4           NaN         NaN        NaN        NaN  ...        NaN        NaN   
..          ...         ...        ...        ...  ...        ...        ...   
474         NaN         NaN        NaN        NaN  ...        NaN        NaN   
475         NaN         NaN        NaN        NaN  ...        NaN        NaN   
476         NaN         NaN        NaN        NaN  ...        NaN        NaN   
477         NaN         NaN        NaN        NaN  ...        NaN        NaN   
478         NaN         NaN        NaN        NaN  ...        NaN        NaN   

    28-09-2013 30-09-2013  10-02-2013  10-04-2013  10-06-2013  10-08-2013  \
0          NaN        NaN         NaN         NaN         NaN         NaN   
1          NaN        NaN         NaN         NaN         NaN         NaN   
2          NaN        NaN         NaN         NaN         NaN         NaN   
3          NaN        NaN         NaN         NaN         NaN         NaN   
4          NaN        NaN         NaN         NaN         NaN         NaN   
..         ...        ...         ...         ...         ...         ...   
474        NaN        NaN         NaN         NaN         NaN         NaN   
475        NaN        NaN         NaN         NaN         NaN         NaN   
476        NaN        NaN         NaN         NaN         NaN         NaN   
477        NaN        NaN         NaN         NaN         NaN         NaN   
478        NaN        NaN         NaN         NaN         NaN         NaN   

     10-10-2013  10-12-2013  
0           NaN         NaN  
1           NaN         NaN  
2           NaN         NaN  
3           NaN         NaN  
4           NaN         NaN  
..          ...         ...  
474         NaN         NaN  
475         NaN         NaN  
476         NaN         NaN  
477         NaN         NaN  
478         NaN         NaN  

[479 rows x 24 columns]

In [ ]:
# Convert the object type columns in "Dress Sales" into float type of data type.


When you see the null counts in “Dress Sales” dataset after performing all the operations that have been mentioned in jupyter notebook, you will find that there are some columns in “Dress Sales” data where there are more than 40% of missing values. Based on your understanding of dealing with missing values do the following steps.

In [ ]:
# Print the null percetange of each column of inp1.


In [ ]:
# Drop the columns in "Dress Sales" which have more than 40% of missing values.


You should categorise the dates into seasons in “Dress Sales” data to simplify the analysis according to the following criteria:
- June, July and August: Summer.
- September, October and November: Autumn.
- December, January and February: WInter.
- March, April and May: Spring.




In [ ]:
# Create the four seasons columns in inp1, according to the above criteria.


In [ ]:
# calculate the sum of sales in each seasons in inp1 i.e. "Dress Sales".


Now let's merge inp1 with inp0 with left join manner, so that the information of inp0 should remain intact.

In [ ]:
# Merge inp0 with inp1 into inp0. this is also called left merge.
inp0 = pd.merge(left=inp0,right=inp1, how='left', left_on='Dress_ID', right_on='Dress_ID')
inp.head()

In [ ]:
# Now Drop the Date columns from inp0 as it is already combined into four seasons.
inp0.drop(inp0.loc[:,'29-08-2013':'10-12-2013'].columns, axis= 1, inplace= True)
inp0.isnull().sum()

Print the null count of inp0 to get the idea about the missing values in data set.

In [ ]:
# Print the null count of each columns in inp0 dataframe i.e. combined data frame of inp0 and inp1 without date columns.


You can see that there are two types of variables one with a large number of missing values and another is very less number of missing values. These two columns can be categorized as:

Type-1: Missing values are very less (around 2 or 3 missing values): Price, Season, NeckLine, SleeveLength, Winter and Autumn. 

Type-2: Missing values are large in numbers (more than 15%): Material, FabricType, Decoration and Pattern Type.



In [ ]:
# Deal with the missing values of Type-1 columns: Price, Season, NeckLine, SleeveLength, Winter and Autumn.


In [ ]:
# Deal with the missing values for Type-2 columns: Material, FabricType, Decoration and Pattern Type.


### Standardise value 

In the given dataset, there are certain discrepancies with the categorical names such as irregular spellings. Choose the correct option of columns with irregular categories and update them.
 
- Season, NeckLine
- Price, Material
- fabricType, Decoration
- Season, SleeveLength


In [ ]:
#correcting the spellings.


In [ ]:
#correcting the Spellings.


## Univariate Analysis 

### Categorical Unordered Univariate Analysis
 

There is a column named ‘Style’ in ‘Attribute Dataset’ which consists of the different style categories of the women apparels. Certain categories whose total sale is less than 50000 across all the seasons is considered under one single category as ‘Others’.


Which of the following categories in ‘Style’ column can be grouped into ‘Others’ category? and perform the grouping operation in the notebook for further analysis.
- Flare, fashion
- Novelty, bohemian
- OL, fashion, work
- Novelty, fashion, Flare


In [52]:
inp0.Style.value_counts()
inp0.head()

Dress_ID    Style    Price  Rating Size  Season NeckLine SleeveLength  \
0  1006032852     Sexy      Low     4.6    M  Summer   o-neck    sleevless   
1  1212192089   Casual      Low     0.0    L  Summer   o-neck        Petal   
2  1190380701  vintage     High     0.0    L  Automn   o-neck         full   
3   966005983    Brief  Average     4.6    L  Spring   o-neck         full   
4   876339541     cute      Low     4.5    M  Summer   o-neck    butterfly   

        Material FabricType  Decoration Pattern Type  Recommendation  
0            NaN    chiffon     ruffles       animal               1  
1     microfiber        NaN     ruffles       animal               0  
2       polyster        NaN         NaN        print               0  
3           silk    chiffon  embroidary        print               1  
4  chiffonfabric    chiffon         bow          dot               0

In [92]:
inp1.describe()

Dress_ID   29-08-2013   31-08-2013   09-02-2013   09-04-2013  \
count  4.790000e+02   479.000000   479.000000   479.000000   479.000000   
mean   9.022420e+08   198.085595   209.776618   223.551148   299.791232   
std    1.822352e+08   579.189322   590.836166   603.098222   601.716515   
min    1.234568e+08     0.000000     0.000000     0.000000     1.000000   
25%    7.666611e+08     0.000000     0.000000     0.000000    28.500000   
50%    9.096250e+08     2.000000     3.000000     4.000000   110.000000   
75%    1.039684e+09   138.500000   165.500000   194.500000   308.500000   
max    1.253973e+09  7455.000000  7467.000000  7479.000000  7374.000000   

        09-06-2013   09-08-2013   09-10-2013   24-09-2013   26-09-2013  \
count   479.000000   479.000000   479.000000   479.000000   257.000000   
mean    304.745303   316.534447   320.100209   372.939457   295.501946   
std     603.854257   609.070537   610.360681   631.674995   696.941427   
min       1.000000     1.000000     1.000000     0.000000     0.000000   
25%      31.500000    36.000000    37.000000    53.000000    19.000000   
50%     116.000000   124.000000   129.000000   178.000000    60.000000   
75%     319.500000   334.000000   334.000000   435.000000   227.000000   
max    7351.000000  7255.000000  7240.000000  6644.000000  6528.000000   

        28-09-2013   30-09-2013   10-02-2013   10-04-2013   10-06-2013  \
count   479.000000   222.000000   220.000000   221.000000   479.000000   
mean    389.590814   240.914414   247.572727   251.058824   415.340292   
std     646.989727   697.151163   707.881500   713.630310   666.827441   
min       0.000000     0.000000     0.000000     0.000000     0.000000   
25%      55.500000    13.250000    13.750000    14.000000    62.000000   
50%     186.000000    52.500000    53.000000    52.000000   200.000000   
75%     465.500000   112.750000   112.250000   111.000000   489.000000   
max    6476.000000  6327.000000  6285.000000  6142.000000  6049.000000   

        10-08-2013   10-10-2013   10-12-2013  
count   224.000000   224.000000   479.000000  
mean    258.437500   262.611607   434.048017  
std     724.092886   732.867748   684.146593  
min       0.000000     0.000000     0.000000  
25%      14.750000    14.750000    65.000000  
50%      57.000000    59.500000   216.000000  
75%     131.250000   133.500000   526.500000  
max    5912.000000  5862.000000  5753.000000

In [110]:
inp1.Dress_ID = inp1.Dress_ID.astype(int)
inp1

Dress_ID  29-08-2013  31-08-2013  09-02-2013  09-04-2013  09-06-2013  \
0    1006032852        2114        2274        2491        2660        2727   
1    1212192089         151         275         570         750         813   
2    1190380701           6           7           7           7           8   
3     966005983        1005        1128        1326        1455        1507   
4     876339541         996        1175        1304        1396        1432   
..          ...         ...         ...         ...         ...         ...   
474   990559192           0           0           0          60          62   
475   713391965           0           0           0         560         554   
476   532874347           0           0           0         734         728   
477   655464934           0           0           0         254         259   
478   919930954           0           0           0         538         545   

     09-08-2013  09-10-2013 09-12-2013 14-09-2013  ... 24-09-2013 26-09-2013  \
0          2887        2930       3119       3204  ...       3554     3624.0   
1          1066        1164       1558       1756  ...       2710     2942.0   
2             8           9         10         10  ...         11       11.0   
3          1621        1637       1723       1746  ...       1878     1892.0   
4          1559        1570       1638       1655  ...       2032     2156.0   
..          ...         ...        ...        ...  ...        ...        ...   
474          64          65         67         68  ...         73       74.0   
475         544         537        525        519  ...        400      388.0   
476         726         715        694        690  ...        616      597.0   
477         261         263        268        270  ...        257      256.0   
478         558         563        578        585  ...        628      632.0   

    28-09-2013 30-09-2013  10-02-2013  10-04-2013  10-06-2013  10-08-2013  \
0         3706     3746.0      3795.0      3832.0        3897      3923.0   
1         3258     3354.0      3475.0      3654.0        3911      4024.0   
2           11       11.0        11.0        11.0          11        11.0   
3         1914     1924.0      1929.0      1941.0        1952      1955.0   
4         2252     2312.0      2387.0      2459.0        2544      2614.0   
..         ...        ...         ...         ...         ...         ...   
474         75       75.0        76.0        76.0          77        77.0   
475        360      364.0       372.0       377.0         380       382.0   
476        586      569.0       561.0       555.0         551       546.0   
477        255      254.0       253.0       250.0         249       249.0   
478        639      645.0       651.0       655.0         660       668.0   

     10-10-2013  10-12-2013  
0        3985.0        4048  
1        4125.0        4277  
2          11.0          11  
3        1959.0        1963  
4        2693.0        2736  
..          ...         ...  
474        77.0          77  
475       384.0         285  
476       535.0         520  
477       249.0         248  
478       674.0         680  

[479 rows x 24 columns]

In [113]:
merged_df = pd.merge(inp0, inp1, on = 'Dress_ID')
merged_df

Dress_ID    Style    Price  Rating  Size  Season   NeckLine  \
0    1006032852     Sexy      Low     4.6     M  Summer     o-neck   
1    1212192089   Casual      Low     0.0     L  Summer     o-neck   
2    1190380701  vintage     High     0.0     L  Automn     o-neck   
3     966005983    Brief  Average     4.6     L  Spring     o-neck   
4     876339541     cute      Low     4.5     M  Summer     o-neck   
..          ...      ...      ...     ...   ...     ...        ...   
474   990559192    Brief  Average     4.7     M  winter     o-neck   
475   713391965   Casual      Low     4.7     M  Spring     o-neck   
476   532874347   Casual  Average     4.7     M  Summer     v-neck   
477   655464934   Casual  Average     4.6     L  winter  boat-neck   
478   919930954   Casual      Low     4.4  free  Summer     v-neck   

    SleeveLength       Material  FabricType  ... 24-09-2013 26-09-2013  \
0      sleevless            NaN     chiffon  ...       3554     3624.0   
1          Petal     microfiber         NaN  ...       2710     2942.0   
2           full       polyster         NaN  ...         11       11.0   
3           full           silk     chiffon  ...       1878     1892.0   
4      butterfly  chiffonfabric     chiffon  ...       2032     2156.0   
..           ...            ...         ...  ...        ...        ...   
474   halfsleeve        acrylic     chiffon  ...         73       74.0   
475         full       polyster         NaN  ...        400      388.0   
476         full         cotton         NaN  ...        616      597.0   
477    sleevless           silk  broadcloth  ...        257      256.0   
478        short         cotton    Corduroy  ...        628      632.0   

     28-09-2013  30-09-2013  10-02-2013  10-04-2013  10-06-2013  10-08-2013  \
0          3706      3746.0      3795.0      3832.0        3897      3923.0   
1          3258      3354.0      3475.0      3654.0        3911      4024.0   
2            11        11.0        11.0        11.0          11        11.0   
3          1914      1924.0      1929.0      1941.0        1952      1955.0   
4          2252      2312.0      2387.0      2459.0        2544      2614.0   
..          ...         ...         ...         ...         ...         ...   
474          75        75.0        76.0        76.0          77        77.0   
475         360       364.0       372.0       377.0         380       382.0   
476         586       569.0       561.0       555.0         551       546.0   
477         255       254.0       253.0       250.0         249       249.0   
478         639       645.0       651.0       655.0         660       668.0   

     10-10-2013  10-12-2013  
0        3985.0        4048  
1        4125.0        4277  
2          11.0          11  
3        1959.0        1963  
4        2693.0        2736  
..          ...         ...  
474        77.0          77  
475       384.0         285  
476       535.0         520  
477       249.0         248  
478       674.0         680  

[479 rows x 36 columns]

In [ ]:
merged_df

In [134]:
# Group "Style" categories into "Others" which have less than 50000 sales across all the seasons.

style_total_sales = merged_df.groupby('Style').sum().sum(axis=1)
styles_to_group = style_total_sales[style_total_sales < 50000].index
merged_df['Categorized_Style'] = merged_df['Style'].apply(lambda x: 'Others' if x in styles_to_group else x)
categorized_sales = merged_df.groupby('Categorized_Style').sum().sum(axis=1)
print(categorized_sales)

Categorized_Style
Brief       1.491326e+10
Casual      1.945748e+11
Flare       2.011578e+09
Novelty     6.799585e+09
OL          1.194627e+09
Sexy        6.121577e+10
bohemian    2.093414e+10
cute        3.949463e+10
fashion     8.367890e+08
party       4.330177e+10
sexy        6.479531e+09
vintage     2.474576e+10
work        1.567372e+10
dtype: float64


In [143]:
merged_df.isna().sum()


Dress_ID               0
Style                  0
Price                  2
Rating                 0
Size                   0
Season                 2
NeckLine               3
SleeveLength           2
Material             119
FabricType           256
Decoration           224
Pattern Type         102
Recommendation         0
29-08-2013             0
31-08-2013             0
09-02-2013             0
09-04-2013             0
09-06-2013             0
09-08-2013             0
09-10-2013             0
09-12-2013             0
14-09-2013             0
16-09-2013             0
18-09-2013             0
20-09-2013             0
22-09-2013             0
24-09-2013             0
26-09-2013           222
28-09-2013             0
30-09-2013           257
10-02-2013           259
10-04-2013           258
10-06-2013             0
10-08-2013           255
10-10-2013           255
10-12-2013             0
Categorized_Style      0
dtype: int64

In [139]:
merged_df.fillna({'Price': merged_df['Price'].mean(),
                  'SleeveLength': merged_df['SleeveLength'].mode()[0],
                 ''}, inplace=True)

0

What is the percentage of “cute” and “Others” category in “Style” column in “Attribute DataSet” respectively?
- 46%, 5%
- 9%, 2.1%
- 2.1%, 5%
- 13.8%, 9%


In [ ]:
# Calculate the percentage of each categories in the "Style" variable.


Similarly Club Neckline, SLeeve length categories into "Others" which have less than 50000 sales across all the seasons.

In [ ]:
# Group "Neckline" categories into "Others" which have less than 50000 sales across all the seasons.


In [ ]:
# Group "Sleeve length" categories into "Others" which have less than 50000 sales across all the seasons.


Club material, fabrictype, patterntype and decoration categories into "Others" which have less than 25000 sales across all the seasons

In [ ]:
# Group "material" categories into "Others" which have less than 25000 sales across all the seasons.


In [ ]:
# Group "fabric type" categories into "Others" which have less than 25000 sales across all the seasons.


In [ ]:
# Group "patern type" categories into "Others" which have less than 25000 sales across all the seasons.


In [ ]:
# Group "decoration" categories into "Others" which have less than 25000 sales across all the seasons.


### Caregorical Ordered Univariate Analysis

Which of the following is an unordered variable in “Attribute DataSet”.
- Style
- Price
- Season
- Size


### Numerical variable Univariate analysis:

What is the approximate difference between the maximum value and 75th percentile in “Autumn” column.
- Approx 54000
- Approx 55000
- Approx 52000
- Approx 50000



In [ ]:
# Describe the numerical variale: "Autumn".


In [ ]:
# plot the boxplot of "Autumn" column.


Which of the following season has the highest difference between the maximum value and 99th quantile of sales?
- Winter
- Summer
- Spring
- Autumn


In [ ]:
# Find the maximum and 99th percentile of Winter season.


In [ ]:
# Find the maximum and 99th percentile of Summer season.


In [ ]:
# Find the maximum and 99th percentile of Spring season.


In [ ]:
# Find the maximum and 99th percentile of Autumn season.


## Bivariate Analysis 

### Numerical- Categorical analysis

Which of the following “Price” category has the lowest average value of rating?
- very-high
- Medium
- Low
- High


In [ ]:
# Find the Mean of Ratings for each Price category.


What is the median of the rating of “vintage” category in Style column?
- 4.6
- 4.7
- 4.55
- 0.00


In [ ]:
# Find the median of Ratings for each Style category.


Which of the following season has the highest average value of sale for “Recommendation” value equals to 1.
- Summer
- Spring
- Autumn
- Winter


In [ ]:
# Summer sale vs Recommendation.


In [ ]:
# Spring sale vs Recommendation.


In [ ]:
# Autumn sale vs Recommendation.


In [ ]:
# Winter sale vs Recommendation.


### Categorical categorical bivariate analysis


Which of the following size categories has the highest positive recommendations?
- Medium and extra large
- Extra large and small
- Free and small
- Free and medium


In [ ]:
# Size vs Recommendation.


### Multivariate analysis 

Which of the following pair of “Style” and “Price” category has the highest average of positive recommendations?
- Price: medium and style: vintage
- Price: medium and style: cute
- Price: very high and style: party
- Price: low and style: sexy


In [ ]:
# plot the heat map of Style, price and Recommendation.


Which of the following material type has no recommendation in summer and winter seasons?
- Mix and Milksilk
- Nylon and Rayon
- Microfiber and Silk
- Milksilk and Microfiber


In [ ]:
# plot the heat map of Season, material and Recommendation.
